src/be/app.py

```diff
@@ -1,14 +1,18 @@
 from pathlib import Path
 import fastapi as fa
 from pydantic import BaseModel
+from yeelight import Bulb, discover_bulbs
+
 
 app = fa.FastAPI()
 
 _fe_dir = Path(__file__).parent.parent / "fe"
 
-
 class Status(BaseModel):
     bulb_name: str
+    power_status: str
+    brightness: int
+    colour: int
```

Comment:
**warning**: big unnecessary write-up incoming 

I'm not sure how colour is represented in yeelight - if it's as an integer then this is allgood.

FYI I remember now - it is an integer, and the way this works for the yeelight colour is:

* The integer value of colour is the integer representation of a hex colour spec
* integers are generally numbers represented with  with base 10 aka decimal (i.e. 0,1,2,3,4,5,6,7,8,9,10,11,...)
* hex numbers have base 16 (i.e. 0,1,2,3,4,5,6,7,8,9,A,B,C,D,E,F,10,11....)
* binary is base 2 (i.e. 0,1,10,11,100,....)
* Look up hex <-> integer conversion in python to understand how
* A Hex colour spec looks like this: 0xFFA403
    * the first two digits (8 bits) FF (255 as decimal) is the R (red) value of RGB
    * the 3rd and 4th digits A4 (164 as decimal) is the G (green) value of RGB
    * the 5th and 6th digits 03 (3 as decimal) is the B (blue) value of RGB

So you could encode it (in the status class) as a tuple of 3 integers (r,g,b) or you could save it as the original int, or as a hex string (i.e. a string like "FFA403"

All options are fine, here is the trade-off (imo) of each

* Tuple (r,g,b)
    * this is going to be easiest to interpret in the front end (therefore more python and less js)
    * will require complexity on the backend (which is better than on the frontend)
* integer as is
    * easiest to implement right now
    * incurs some tech debt, as you will just need to process this later (unless this representation is supported in other places, it might be, but I haven't really seen it)
* hex string
    * kinda nicer to read? but not really
    * strings are simple to transmit
    * probably going to be complex on the fe and be tbh

Reply:
if you're struggling to decide, just huck in the integer now an not worry about it haha